### PR TITLE
net: refactor named pipe builders to not use bitfields

### LINKED ISSUE #5359: Make named pipe configuration less error prone
The current design of the named pipe builder resulted in [CVE-2023-22466][1]. It should be refactored to make this kind of mistake impossible.

Refs: #5336

[1]: https://github.com/tokio-rs/tokio/security/advisories/GHSA-7rrj-xr53-82p7

### PR DESCRIPTION
**Pull request by mhils**
This fixes #5359.

### CODE REVIEW

File: tokio/src/net/windows/named_pipe.rs [hunk 1]
```diff
@@ -1687,8 +1683,14 @@ impl ServerOptions {
     /// ```
     pub fn new() -> ServerOptions {
         ServerOptions {
-            open_mode: windows_sys::PIPE_ACCESS_DUPLEX | windows_sys::FILE_FLAG_OVERLAPPED,
-            pipe_mode: windows_sys::PIPE_TYPE_BYTE | windows_sys::PIPE_REJECT_REMOTE_CLIENTS,
```
Comment (Darksonn) [inline tokio/src/net/windows/named_pipe.rs:1691]: The `PIPE_ACCESS_DUPLEX` option seems to have disappeared?

Comment (mhils) [reply tokio/src/net/windows/named_pipe.rs:1691]: [`PIPE_ACCESS_DUPLEX`](https://learn.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-createnamedpipea) is `PIPE_ACCESS_INBOUND | PIPE_ACCESS_OUTBOUND`.

**Review by Darksonn — APPROVED** [review]
Ok, LGTM.

**Comment by Darksonn** [comment]
Thanks!

**Pull request merged**